### PR TITLE
LF-3266 - Enhancers | Inactive users: display 

### PR DESCRIPTION
**Pull request by thxv3n0lvl**
**Description**

Added a couple validations to TaskCard so when a revoked user still assigned to a task, their name will be displayed with a "(inactive)" mark and also greys out the icon and name. This was an issue in terms of not handling the behavior described in the ticket  but it was discussed in the same ticket this should be the behavior although I might be loosing some bits of the conversation.

Jira link:
https://lite-farm.atlassian.net/browse/LF-3266

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

3 scenarios were tested on the Task view:
1. Task assigned to an active user -> name displays as always
1. Task unassigned -> unassigned format displays as always
1. Task assigned to inactive user -> new format in place

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [x] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

Screenshot of the implementation:
image
image

### CODE REVIEW

**Review by Duncan-Brain — APPROVED** [review]
Sorry it took a while to respond Alan!

Everything looks great. Not sure whether the optional chaining operator (ie assignee?.status) could have saved a line or two of code or not. But that is the only thing I might've considered.

Thanks!

**Comment by thxv3n0lvl** [comment]
> Sorry it took a while to respond Alan!
> 
> Everything looks great. Not sure whether the optional chaining operator (ie assignee?.status) could have saved a line or two of code or not. But that is the only thing I might've considered.
> 
> Thanks!

@Duncan-Brain correct me if I'm wrong but I think if you refer to do  this instead
`const isAssigneeInactive = assignee?.status === 'Inactive';`
and removing the if condition `if(assignee !== null)` it will lead to an error as the assignee object is actually null when the "Unassigned" option is selected for the task. Let me know if this is not what you mean.

On the other hand, I'm not sure how to proceed with this PR given [this](https://litefarm.slack.com/archives/C041EMXTBPY/p1688407643866929) conversation.

**Comment by Duncan-Brain** [comment]
@thxv3n0lvl 

I see.. it looks like we had a duplicate issue haha That can be confusing sorry about that! 

Let me see if we can clear that up on the other ticket. It seems pretty small to reuse so I will inquire further. But I think in the meantime we could merge and they could always update this implementation with a reusable component once that is made.

As for the null condition -- really don't worry about it haha.  It is probably more readable as-is. Its just fun for me to think about how few lines of code could be used. Here is a little test of what I believed would happen in that case:
Screenshot 2023-07-05 at 9 26 29 AM

**Review by Duncan-Brain — CHANGES_REQUESTED** [review]
Maybe this is actually more full stack than I realized! Re-reading again says maybe two other things could be completed:

watching the video you can see the task is unassigned when clicked -- presumably we want to keep the task assigned to the inactive user I am getting from the conversation around it
Also in finances section under labour expenses the same (inactive) should be applied*

File: packages/webapp/src/components/CardWithStatus/TaskCard/TaskCard.jsx [hunk 1]
```diff
@@ -67,6 +67,14 @@ export const PureTaskCard = ({
     onClickCompleteOrDueDate?.();
   };
 
+  const isAssigneeInactive = assignee?.status === 'Inactive';
+  let assigneeName = '';
+  if (assignee !== null) {
```
Comment (thxv3n0lvl) [inline packages/webapp/src/components/CardWithStatus/TaskCard/TaskCard.jsx:72]: @Duncan-Brain I left the null validation just to make sure we don't miss the unassigned state of a task. Let me know if you would llike to see this one differently.

File: packages/webapp/src/components/CardWithStatus/TaskCard/TaskCard.jsx [hunk 2]
```diff
@@ -67,6 +67,14 @@ export const PureTaskCard = ({
...
+    const lastName =
+      assignee.last_name.length > 0 ? assignee.last_name.toUpperCase().charAt(0) + '.' : '';
+    assigneeName = `${assignee.first_name} ${lastName}`;
+  }
+
```
Comment (thxv3n0lvl) [inline packages/webapp/src/components/CardWithStatus/TaskCard/TaskCard.jsx:77]: for this one I left the `lastName`  in a separate constant just to make it legible as there's a bit of logic to present the last name in a "capital first letter and dot" format (not sure how to call it 😂 ). Also is possible that a username doesn't have a last name, so I left a validation for that.

File: packages/webapp/src/components/CardWithStatus/TaskCard/styles.module.scss [hunk 3]
```diff
@@ -88,3 +88,12 @@
...
 }
+
+.inactive {
+  color: var(--grey500);
+}
+
+.firstInitial.inactive {
+  background-color: var(--grey500);
+  color: white;
+}
```
Comment (thxv3n0lvl) [inline packages/webapp/src/components/CardWithStatus/TaskCard/styles.module.scss:99]: @Duncan-Brain I don't have access to figma but maybe someone can confirm the correct color for the inactive label on the task card view.

**Comment by thxv3n0lvl** [comment]
I updated the name of this PR to avoid confusions with the other tickets.

File: packages/webapp/src/components/CardWithStatus/TaskCard/TaskCard.jsx (hunk 1 above)
Comment (Duncan-Brain) [reply packages/webapp/src/components/CardWithStatus/TaskCard/TaskCard.jsx:72]: looks great!

**Review by Duncan-Brain — APPROVED** [review]
Thanks Alan looks great, if there is any chance you could update the Jira link in the description too for the next reviewer that would be cool!

**Pull request merged**